troubleshooting.md

```diff
@@ -0,0 +1,551 @@
+
+The `install-config.yaml` file needs to be generated with proper details else the installations will fail.
+
+1. Make sure this file has the SSH keys of the provisioner node, if not you can't login into any of the nodes after deployment is completed.
+
+2. Make sure you have proper values set for fields for 
+
+```
+bootMACAddress: 24:6E:96:1B:96:90 # MAC of bootable internal NIC
+hardwareProfile: default          #master node settings
+```
+and 
+```
+bootMACAddress: 24:6E:96:1B:96:90 # MAC of bootable internal NIC
+hardwareProfile: unknown          #worker node settings
+```
+
+* Verify the syntax of install-config.yaml file, confirm that you can reach the `bootstrapOSImage` and `clusterOSImage` URL from the Jump Host. 
+
+
+**Important**:  Make sure to power off the master nodes before you begin the installation via iDRAC/iLOM.
+
+Feel free to use this below script:
+
+```
+#!/bin/bash
+
+echo "***Power Off master-0,master1,master-2****"
+ipmitool -I lanplus -U root -P <Password> -H <iDRAC-ip> power off
+ipmitool -I lanplus -U root -P <Password> -H <iDRAC-ip> power off
+ipmitool -I lanplus -U root -P <Password> -H <iDRAC-ip> power off
+``` 
+
+## During the deployment
+
+Create your metal3-config.yaml file as shown below 
+
+```yaml
+ apiVersion: v1
+ kind: ConfigMap
+ metadata:
+   name: metal3-config
+   namespace: openshift-machine-api
+ data:
+   cache_url: rhcos-43.81.202001142154.0-qemu.x86_64.qcow2.gz
+   deploy_kernel_url: http://172.22.0.1:6180/images/ironic-python-agent.kernel
+   deploy_ramdisk_url: http://172.22.0.1:6180/images/ironic-python-agent.initramfs
+   dhcp_range: 172.22.0.10,172.22.0.100
+   http_port: "6180"
+   ironic_endpoint: http://172.22.0.1:6385/v1/
+   ironic_inspector_endpoint: http://172.22.0.3:5050/v1/
+   provisioning_interface: eno1
+   provisioning_ip: 172.22.0.1/24
+   rhcos_image_url: http://[Ipv4/Ipv6-IP-Registry]/rhcos-43.81.202001142154.0-openstack.x86_64.qcow2.gz
+```
+
+Before running the installation, please make sure you have a cleaned the previous files/manifests.
+
+# Bootstrap VM related issues 
+
+After the installation is triggered, Check/verify bootstrap VM is up and running from worker node: 
+
+```
+[kni@provisioner]$ sudo virsh list
+ Id    Name                           State
+----------------------------------------------------
+ 12    kni7-xf6fq-bootstrap           running
+```
+
+If there is no sign of Bootstarp VM running on the provioner node, check if the libvirtd is running(if not, start the service) on the provisioner node. 
+
+```
+systemctl status libvirtd
+● libvirtd.service - Virtualization daemon
+   Loaded: loaded (/usr/lib/systemd/system/libvirtd.service; enabled; vendor preset: enabled)
+   Active: active (running) since Mon 2020-02-03 13:08:51 EST; 2 weeks 0 days ago
+     Docs: man:libvirtd(8)
+```
+
+If re-running the deployment, please make sure to delete all the older VM including its volume as shown below:
```

Comment:
Why? What happens if I leave it? What is the advantage of removing it?

Reply:
Mentioned IP conflict problems if this VM is uncleaned.